### ISSUE TITLE
[BUG][Fuzzer][wrong-code] T.all_of/T.any_of over a swizzle-layout shared buffer silently return wrong results (linear scan ignores the swizzle permutation)

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2714]
## Problem description

`T.all_of`/`T.any_of` applied to a row of a shared buffer that carries a `make_swizzled_layout` silently return the wrong boolean. The lowering emits a scan whose **base address carries the swizzle permutation** but whose **element count drives a plain contiguous scan** (`for i in 0..elems: a[base+i]`), so the scan reads the wrong physical slots — running off the row or into a neighbouring row. The same kernel without the swizzle annotation is correct, so the op is accepted and computes correctly on the default layout; only the swizzle-layout case is mishandled. No crash, no diagnostic.

## Reproducible example code

Two faces of one root — an `all_of` false-negative and an `any_of` cross-row false-positive. Each has a no-swizzle control (correct) in the same shape.

```python
import os, tempfile
d = tempfile.mkdtemp(); os.environ["TILELANG_CACHE_DIR"] = d+"/c"; os.environ["TMPDIR"] = d+"/t"
os.makedirs(os.environ["TMPDIR"], exist_ok=True)
import torch, tilelang, tilelang.language as T
from tilelang.layout import make_swizzled_layout
tilelang.disable_cache()
M, N = 16, 64

def build(op, swizzle):
    @tilelang.jit
    def main(A: T.Tensor((M, N), "float16"), Out: T.Tensor((M,), "bool")):
        with T.Kernel(1, threads=32):
            s = T.alloc_shared((M, N), "float16")
            if swizzle:
                T.annotate_layout({s: make_swizzled_layout(s)})
            T.copy(A, s)
            for r in T.serial(M):
                Out[r] = op(s[r, :])
    return main

# --- all_of on an all-ones buffer: every row is all-nonzero -> all_of must be True everywhere ---
a = torch.ones(M, N, dtype=torch.float16, device="cuda")
for swz in (True, False):
    out = torch.empty(M, dtype=torch.bool, device="cuda"); build(T.all_of, swz)(a, out)
    wrong = [(r, bool(out[r].cpu())) for r in range(M) if not bool(out[r].cpu())]
    print(f"all_of swizzle={swz}: wrong rows {wrong}")

# --- any_of with only row 8 nonzero: any_of must be True for row 8 ONLY ---
a = torch.zeros(M, N, dtype=torch.float16, device="cuda"); a[8, :] = 1.0
for swz in (True, False):
    out = torch.empty(M, dtype=torch.bool, device="cuda"); build(T.any_of, swz)(a, out)
    wrong = [(r, bool(out[r].cpu())) for r in range(M) if bool(out[r].cpu()) != (r == 8)]
    print(f"any_of swizzle={swz}: wrong rows {wrong}")
```

Output on L40S (sm_89), tilelang 0.1.11 (deterministic ×2):

```
all_of swizzle=True:  wrong rows [(15, False)]     <- row 15 all-ones, wrongly reports NOT-all
all_of swizzle=False: wrong rows []                <- correct
any_of swizzle=True:  wrong rows [(7, True)]        <- row 7 all-zero, wrongly reports any (reads row 8)
any_of swizzle=False: wrong rows []                <- correct
```

## Traceback

No traceback — compiles and runs, boolean silently wrong.

## Expected behavior

`all_of`/`any_of` over `s[r, :]` should reduce exactly the logical elements of row `r`, independent of the buffer's physical layout — the no-swizzle control (identical kernel) does this correctly for both ops. The reduction must honour the swizzle permutation the buffer carries (either by reading the permuted element positions, or by scanning the logical row rather than a contiguous physical span).

## System information

tilelang 0.1.11 (repro), source verified on `origin/main` at `c5c10d26`. L40S, sm_89, CUDA 12.x. Cache disabled, private `TILELANG_CACHE_DIR`/`TMPDIR`.

## Additional context

**Root cause.** The lowering passes the (swizzled) element *base address* and a plain element *count* to a device helper that scans contiguously from the base, so under a swizzle layout the scanned physical slots are not the row's logical elements.

<details>
<summary>Mechanism (pinned to <code>c5c10d26</code>) — the emitted CUDA is the smoking gun</summary>

The emitted CUDA for `all_of(s[r, :])` over the swizzle buffer is:

```cuda
Out[r] = ((signed char)tl::All((&(s[((r * 64) + ((r & 7) * 8))])), 64));
```

The base index `r*64 + (r&7)*8` **carries the swizzle permutation** (the `(r&7)*8` XOR-style term), but the count `64` is the plain row width.

- [`all_of_op`/`any_of_op` in `logical.cc` L16-35](https://github.com/tile-ai/tilelang/blob/c5c10d26c2215692ef133113362364b5fd9494cc/src/op/logical.cc#L16-L35) build `tl::All`/`tl::Any(buffer_address, elems)` where `buffer_address` is the swizzled base and `elems` is the logical row length.
- [`tl::All`/`tl::Any` in `common.h` L592-611](https://github.com/tile-ai/tilelang/blob/c5c10d26c2215692ef133113362364b5fd9494cc/src/tl_templates/cuda/common.h#L592-L611) do a **flat contiguous scan** `for (int i = 0; i < size; i++) a[i]`, which assumes an identity layout from the base.

So for row `r`, the scan reads physical slots `base(r) .. base(r)+63` linearly, but the row's logical elements are permuted by the swizzle and are not that contiguous span. Row 15 (`base = 15*64 + 7*8 = 1016`) scans `[1016, 1080)` past the 1024-slot buffer → `all_of` sees stale/OOB slots → wrong `False`. In the `any_of` case row 7 (`base = 7*64 + 7*8 = 504`) scans `[504, 568)`, which overlaps row 8's physical region `[512, 576)` → reads row 8's nonzero data → wrong `True`. The no-swizzle layout makes `base(r) = r*64` with the row genuinely contiguous, so the same scan is correct (matches the control).
</details>

**Provenance.** Present on `origin/main` at `c5c10d26` (verified by reading `logical.cc`/`common.h` there). Reproduced this session (deterministic ×2) on tilelang 0.1.11 / L40S sm_89; emitted CUDA dumped via `get_kernel_source`. Source lines pinned to `c5c10d26`; permalinks curl-checked (HTTP 200).

**Dedup.** No filed issue references `any_of`/`all_of` value-fidelity under a swizzle layout. Distinct from the filed swizzle-span issues (#2555/#2578/#2593/#2603, which are TMA descriptor/launch-span or Hopper linear-fallback cases) and from #2609 (view+swizzle reshape rescale) — those are different ops and mechanisms; this is the `logical.cc` reduction lowering feeding a layout-unaware contiguous scan.

**Reach.** `T.all_of`/`T.any_of` over a buffer region and `make_swizzled_layout` are both documented, supported APIs; combining them (e.g. a boolean mask/predicate check over a swizzled tile that is also used as a GEMM/MMA operand, where swizzle avoids bank conflicts) is a natural pattern. The reduction is accepted, compiled, and run, and returns a wrong boolean with no diagnostic. Both ops share the one root (contiguous scan of a swizzled base); the two faces above (row-truncation false-negative and cross-row false-positive) are the same defect.
